lib/tmuxinator/cli.rb

```diff
@@ -69,25 +71,86 @@ def completions(arg)
                           aliases: ["-l"],
                           desc: "Create local project file at ./.tmuxinator.yml"
 
-    def new(name)
-      project_file = find_project_file(name, options[:local])
-      Kernel.system("$EDITOR #{project_file}") || doctor
+    def new(name, session = nil)
+      if session
+        new_project_with_session(name, session)
+      else
+        new_project(name)
+      end
     end
 
     no_commands do
+      def new_project(name)
+        project_file = find_project_file(name, options[:local])
+        Kernel.system("$EDITOR #{project_file}") || doctor
+      end
+
+      def new_project_with_session(name, session)
```

Comment:
There's a fair amount that's going on here...  Would be helpful to have a couple of unit-test specs to constrain the functionality that this represents.